src/marshmallow/validate.py

```diff
@@ -170,41 +170,66 @@ def __call__(self, value):
 
 
 class Range(Validator):
-    """Validator which succeeds if the value it is passed is greater
-    or equal to ``min`` and less than or equal to ``max``. If ``min``
-    is not specified, or is specified as `None`, no lower bound
-    exists. If ``max`` is not specified, or is specified as `None`,
-    no upper bound exists.
+    """Validator which succeeds if the value passed to it is within the specified
+    range. If ``min`` is not specified, or is specified as `None`,
+    no lower bound exists. If ``max`` is not specified, or is specified as `None`,
+    no upper bound exists. The inclusivity of the bounds (if they exist) is configurable.
+    If ``min_inclusive`` is not specified, or is specified as `True`, then
+    the ``min`` bound is included in the range. If ``max_inclusive`` is not specified,
+    or is specified as `True`, then the ``max`` bound is included in the range.
 
     :param min: The minimum value (lower bound). If not provided, minimum
         value will not be checked.
     :param max: The maximum value (upper bound). If not provided, maximum
         value will not be checked.
     :param str error: Error message to raise in case of a validation error.
         Can be interpolated with `{input}`, `{min}` and `{max}`.
+    :param bool min_inclusive: Whether the `min` bound is included in the range.
+    :param bool max_inclusive: Whether the `max` bound is included in the range.
     """
 
-    message_min = 'Must be at least {min}.'
-    message_max = 'Must be at most {max}.'
-    message_all = 'Must be between {min} and {max}.'
+    message_min = 'Must be {min_op} {{min}}.'
+    message_max = 'Must be {max_op} {{max}}.'
+    message_all = 'Must be {min_op} {{min}} and {max_op} {{max}}.'
 
-    def __init__(self, min=None, max=None, *, error=None):
+    message_gte = 'greater or equal to'
+    message_gt = 'greater than'
+    message_lte = 'less or equal to'
+    message_lt = 'less than'
+
+    def __init__(self, min=None, max=None, min_inclusive=True, max_inclusive=True, *, error=None):
```

Comment:
`min_inclusive` and `max_exclusive` should be keyword-only.

```suggestion
    def __init__(self, min=None, max=None, *, min_inclusive=True, max_inclusive=True, error=None):
```